cmd/mdatagen/internal/metadata.go

```diff
@@ -338,3 +343,52 @@ func (a Attribute) TestValue() string {
 	}
 	return ""
 }
+
+func (a Attribute) TestValueTemplateKey() string {
+	switch a.Type.ValueType {
+	case pcommon.ValueTypeEmpty:
+		return ""
+	case pcommon.ValueTypeStr:
+		return `"key", "val"`
+	case pcommon.ValueTypeInt:
+		return `"key", 1`
+	case pcommon.ValueTypeDouble:
+		return `"key", 1.1`
+	case pcommon.ValueTypeBool:
+		return `"key", true`
+	case pcommon.ValueTypeMap:
+		return `"key", map[string]any{"key1": "val1", "key2": "val2"}`
+	case pcommon.ValueTypeSlice:
+		return `"key", []any{"item1", "item2"}`
+	case pcommon.ValueTypeBytes:
+		return `"key", []byte("val")`
+	}
+	return ""
+}
+
+func (a Attribute) TestResultTemplateValue() string {
```

Comment:
Could we share code between this function and `TestValue` above so this function returns real values, and to reduce some of the apparent duplication?